### PR TITLE
Added Docker support

### PR DESCRIPTION
**Pull request by nefarius**
**Please describe the changes this PR makes and why it should be merged:**

Added files required to build and run this bot with Docker (see #385) and an example stack with containerised Lavalink server.

**Status and versioning classification:**

- This PR **only** includes non-code changes, like changes to documentation, README, etc.

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -0,0 +1,11 @@
+FROM node:14-alpine
```
Comment (SudhanPlayz) [inline Dockerfile:1]: uhm can we use node v16

File: Dockerfile [hunk 2]
```diff
@@ -0,0 +1,11 @@
...
+
+WORKDIR /usr/src/app
+
+COPY package*.json ./
+
+RUN npm install
```
Comment (SudhanPlayz) [inline Dockerfile:7]: what about yarn its better than npm

File: docker-compose.yml.example [hunk 3]
```diff
@@ -0,0 +1,27 @@
+version: "3"
+
+services:
+  music-bot:
+    build: .
+    image: music-bot:latest
```
Comment (SudhanPlayz) [inline docker-compose.yml.example:6]:
```suggestion
    image: discord-musicbot:latest
```

File: docker-compose.yml.example [hunk 4]
```diff
@@ -0,0 +1,27 @@
+version: "3"
+
+services:
+  music-bot:
```
Comment (SudhanPlayz) [inline docker-compose.yml.example:4]:
```suggestion
  discord-musicbot:
```

File: docker-compose.yml.example [hunk 5]
```diff
@@ -0,0 +1,27 @@
...
+    build: .
+    image: discord-musicbot:latest
+    container_name: music-bot
```
Comment (SudhanPlayz) [inline docker-compose.yml.example:7]:
```suggestion
    container_name: discord-musicbot
```

File: docker-compose.yml.example (hunk 4 above)
Comment (SudhanPlayz) [inline docker-compose.yml.example]:
```suggestion
  discord-musicbot:
```

**Review by SudhanPlayz — CHANGES_REQUESTED** [review]
Need changes

File: Dockerfile (hunk 1 above)
Comment (nefarius) [reply Dockerfile:1]: Can do but now throws:

```node
npm WARN EBADENGINE Unsupported engine {
npm WARN EBADENGINE   package: 'discord-musicbot@4.1.1',
npm WARN EBADENGINE   required: { node: '14.x' },
npm WARN EBADENGINE   current: { node: 'v16.9.1', npm: '7.21.1' }
npm WARN EBADENGINE }
```

File: Dockerfile (hunk 2 above)
Comment (nefarius) [reply Dockerfile:7]: I took the example from your Wiki and within Docker does it really make a difference? It's only executed upon build anyway.

Comment (nefarius) [reply Dockerfile:7]: `yarn` build hard fails when using Node 16:

```node
error discord-musicbot@4.1.1: The engine "node" is incompatible with this module. Expected version "14.x". Got "16.9.1"
```

**Review by VinceBT — APPROVED** [review]

**Pull request merged**